### PR TITLE
DSD-1149: Adding button style variants for the Link component

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1149](https://jira.nypl.org/browse/DSD-1149)

## This PR does the following:

- Adds button-styled variants for the `Link` component through the `type` prop. This includes `"buttonPrimary"`, `"buttonSecondary"`, `"buttonPill"`, `"buttonCallout"`, `"buttonNoBrand"`, and `"buttonDisabled"`.
- Deprecates the existing `"button"` type.

## How has this been tested?

Locally in Storybook.

## Accessibility concerns or updates

- N/A although more research will be done for further updates.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Review by jackiequach — COMMENTED** [review]
The hover states do not match `Button`'s styles.

**Comment by EdwinGuzman** [comment]
@jackiequach which hover states don't match? The underline under the text when hovering is intentional. I think the rest of the styles match their button counterpart.

**Comment by jackiequach** [comment]
@EdwinGuzman if that is the case, there is no issue then.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 ### Adds
 
 - Adds a variant for `NYCDOE` to the `Logo` component.
+- Adds the `"buttonPrimary"`, `"buttonSecondary"`, `"buttonPill"`, `"buttonCallout"`, `"buttonNoBrand"`, `"buttonDisabled"` variants for the the `Link` component, set through the `type` prop.
```
Comment (bigfishdesign13) [inline CHANGELOG.md:13]: Add a note about deprecating the `"button"` variant.

File: src/theme/components/link.ts [hunk 2]
```diff
@@ -45,6 +60,43 @@ const variants = {
       textDecoration: "none",
     },
   },
+  buttonPrimary: {
```
Comment (bigfishdesign13) [inline src/theme/components/link.ts:63]: I forgot that we already had a number of reusable style objects in place for `Link`.  Love it.

**Review by bigfishdesign13 — APPROVED** [review]
Looks great.  Just make sure to add somethign in the CHANGELOG about the deprecated "button" variant.

**Pull request merged**